crates/router/src/db/payment_intent.rs

```diff
@@ -307,13 +300,15 @@ mod storage {
                 .into_report()
         }
 
+        #[cfg(feature = "olap")]
         async fn filter_payment_intent_by_constraints(
             &self,
             merchant_id: &str,
-            pc: &api::PaymentListConstraints,
+            pc: &crate::types::api::PaymentListConstraints,
             _storage_scheme: enums::MerchantStorageScheme,
         ) -> CustomResult<Vec<PaymentIntent>, errors::StorageError> {
-            let conn = pg_connection(&self.master_pool).await;
+            #[cfg(feature = "olap")]
```

Comment:
This annotation shouldn't be required since the whole function has the annotation, isn't it?